### PR TITLE
Lock version using pessimistic version operator. Fixes #1930

### LINKED ISSUE #1930: deploy.rb template should use pessimistic version operator for `lock`
Assuming I am running Capistrano 3.9.1, then `cap install` generates the following in `deploy.rb`:

```ruby
lock '3.9.1'
```

Relevant code: https://github.com/capistrano/capistrano/blob/bf45cbafe5ef396933c81f334bdbc8c6d32a38c2/lib/capistrano/templates/deploy.rb.erb#L2

This means that my project can *only* be deployed using version 3.9.1.

This seems unnecessarily strict. If the Capistrano maintainers are doing a good job of adhering to the spirit of semver, then it should also be safe to deploy with 3.9.2, 3.9.3, etc. Arguably, I should be able to use *any* 3.x version, like 3.10, 3.11, etc. (Although Capistrano is only loosely following semver, so this might be a bit risky. See #1894.)

I can achieve this by changing my `deploy.rb` as follows:

```ruby
# Patch releases are OK
lock '~> 3.9.1'
```

Or more broadly:

```ruby
# Feature bumps are OK
lock '~> 3.9'
```

(The `~>` operator is called the "pessimistic version" operator in rubygems and Bundler.)

I would like to make the change to the `deploy.rb` template to use `~>`. The question is, should we lock to the patch level (`x.y.z`) or the minor (`x.y`) level?

### PR DESCRIPTION
**Pull request by will-in-wi**
Fix for https://github.com/capistrano/capistrano/issues/1930

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -114,6 +114,7 @@ gem "capistrano", :github => "capistrano/capistrano"
 
 * Suppress log messages of `git ls-remote` by filtering remote refs (@aeroastro)
 * The Git SCM now allows the repo_url to be changed without manually wiping out the mirror on each target host first (@javanthropus)
+* Default to locking the version using the pessimistic version operator at the patch level.
```
Comment (mattbrictson) [inline CHANGELOG.md]: This should be under the `master` section, right?

Comment (will-in-wi) [reply CHANGELOG.md]: Yup. Fixed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ gem "capistrano", :github => "capistrano/capistrano"
 * Your contribution here!
 * As of this release, version 2.x of Capistrano is officially End of Life. No further releases of 2.x series are planned, and pull requests against 2.x are no longer accepted. The maintainers encourage you to upgrade to 3.x if possible.
 * [#1937](https://github.com/capistrano/capistrano/pull/1937): Clarify error message when plugin is required in the wrong config file.
+* Default to locking the version using the pessimistic version operator at the patch level.
```
Comment (mattbrictson) [inline CHANGELOG.md]: I think it would be helpful to link to #1930 in the CHANGELOG entry. Otherwise this looks good to merge. Thanks!

**Comment by will-in-wi** [comment]
Thanks! I added the link.

**Pull request merged**